src/filesystem/impls/filer/lib/transforms.js

```diff
@@ -0,0 +1,172 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define */
+define(function (require, exports, module) {
+    "use strict";
+
+    var CommandManager = require("command/CommandManager");
+    var Commands = require("command/Commands");
+    var BracketsFiler = require("filesystem/impls/filer/BracketsFiler");
+    var LanguageManager = require("language/LanguageManager");
+    var ExtensionUtils = require("utils/ExtensionUtils");
+    var marked = require("thirdparty/marked/marked.min");
+    var StartupState = require("bramble/StartupState");
+    var Path = BracketsFiler.Path;
+
+    var _transforms = {};
+    var _fs = BracketsFiler.fs();
+
+    function _register(transform) {
+        _transforms[transform.srcExt] = transform;
+    }
+
+    function getTransform(path) {
+        var ext = Path.extname(path).replace(/^\./, ".").toLowerCase();
```

Comment:
don't you need a `?` after the `^\.` in your regex? Otherwise it won't add a `.` if it does not exist no?